src/main/java/org/moe/maven/GradleTask.java

```diff
@@ -166,8 +166,8 @@ public void onComplete(Void arg0) {
 
                 @Override
                 public void onFailure(GradleConnectionException exc) {
-                    completed = true;
                     exception = exc;
+                    completed = true;
```

Comment:
One more change: Changing the order of writes to avoid race condition when reading the `exception` field before it gets assigned.